v0/pipeline.json

```diff
@@ -14489,7 +14489,7 @@
           "Approvers" : {
             "title" : "Approvers",
             "type" : "object",
-            "required" : [ "disallowPipelineExecutor", "minimumCount", "userGroups" ],
+            "required" : [ "disallowPipelineExecutor", "minimumCount" ],
```

Comment:
we are removing a required field, this won't be backward compatible. Same comment in other files too.

Reply:
@rohitk-h We initially had userGroups as required while configuring the Harness Approval Step.
Now we have a feature planned for this quarter to have support of serviceAccountTokens as well for approving/rejecting pipelines. So we cannot keep userGroups as required as customer might only want to configure serviceAccounts to their step. Hence this change is necessary. 
Regarding the backward compatibility, we should be good as we are removing the required field and not adding one. Hence the existing APIs for the users will continue to work without any change in API request payload. 

Tech spec for the feature - https://harness.atlassian.net/wiki/spaces/CDNG/pages/22014755252/Use+Approvals+API+with+Service+Accounts+Auth

Reply:
thanks for the explanation!